lib/pbench/cli/server/shell.py

```diff
@@ -16,14 +18,13 @@
 PROG = "pbench-shell"
 
 
-def app():
+def app() -> Flask:
     """External gunicorn application entry point."""
     try:
-        server_config = get_server_config()
-    except (ConfigFileNotSpecified, BadConfig) as e:
+        return create_app(get_server_config())
+    except Exception as e:
         print(e, file=sys.stderr)
         sys.exit(1)
-    return create_app(server_config)
```

Comment:
While reviewing the `rest_uri` exception contexts where it would be raised, I had searched for the use of `BadConfig`.  I found this case where `get_server_config()` was called and `BadConfig` was used to catch the error, but noticed that `create_app()` was calling `sys.exit(1)` on a database initialization exception.  This change addresses the error handling properly.